plugins/trace_api_plugin/include/eosio/trace_api/request_handler.hpp

```diff
@@ -6,12 +6,13 @@
 #include <eosio/trace_api/common.hpp>
 
 namespace eosio::trace_api {
-   using data_handler_function = std::function<fc::variant(const action_trace_v0&, const yield_function&)>;
-
+   using data_handler_function_v0 = std::function<std::tuple<fc::variant, fc::optional<fc::variant>>(const action_trace_v0&, const yield_function&)>;
```

Comment:
Instead of having these specializations, which will grow to even more as we add more trace versions, you could use an `std::variant` as the first argument.

So, 

``` c++
using data_handler_function = std::function<std::tuple<fc::variant, std::optional<fc::variant>>( const action_tract_t&, const yield_function&)>;
``` 
where `action_trace_t` is the std::variant of the valid action trace versions.